src/driver/test/workload_parsers_test.cpp

```diff
@@ -19,6 +19,51 @@
 namespace genny::driver::v1 {
 namespace {
 
+TEST_CASE("Contexts have scope") {
+    Context* context = nullptr;
+    ContextGuard outerContext(context);
+
+    YAML::Node outer;
+    outer["outerKey"] = "outerVal";
+    context->insert("outerName", outer, Type::kParameter);
+
+    {
+        ContextGuard innerContext(context);
+        YAML::Node inner;
+        inner["innerKey1"] = "innerVal1";
+        context->insert("innerName1", inner, Type::kParameter);
+
+        auto retrievedOuter = context->get("outerName", Type::kParameter);
+        REQUIRE(retrievedOuter == outer);
+
+        auto retrievedInner = context->get("innerName1", Type::kParameter);
+        REQUIRE(retrievedInner == inner);
+    }
+
+    {
+        ContextGuard innerContext(context);
+        YAML::Node inner;
+        inner["innerKey2"] = "innerVal2";
+        context->insert("innerName2", inner, Type::kParameter);
+
+        auto retrievedOuter = context->get("outerName", Type::kParameter);
+        REQUIRE(retrievedOuter == outer);
+
+        auto retrievedInner = context->get("innerName2", Type::kParameter);
+        REQUIRE(retrievedInner == inner);
+
+        auto retrievedOldInner = context->get("innerName1", Type::kParameter);
+        REQUIRE_FALSE(retrievedOldInner == inner);
+
+    }
+
+    auto retrievedOuter = context->get("outerName", Type::kParameter);
+    REQUIRE(retrievedOuter == outer);
+
+    REQUIRE_THROWS(context->get("outerName", Type::kActorTemplate));
```

Comment:
I don't think this test would guard against regressions that could be introduced if we move away from `ContextGuard` to something else. Is there a way to test the scoping functionality using actor yaml directly?

Reply:
Added an actor yaml-based test. I think keeping this test has value too; this one is a lot more explicit than the yaml-based one.